Network/Wreq/Internal.hs

```diff
@@ -117,8 +118,10 @@ prepare modify opts url = do
     signRequest = maybe return f $ auth opts
       where
         f (AWSAuth versn key secret) = AWS.signRequest versn key secret
```

Comment:
[bos wrote] While you're at it, please also fix that API mistake for the AWS code in a separate commit

We're not passing the auth record to AWS.signRequest, but only the constituent parts. What could I do better here?